crates/rpc-types-eth/src/block.rs

```diff
@@ -204,324 +205,6 @@ impl TryFrom<Header> for alloy_consensus::Header {
-        BlockTransactionHashesMut::new(self)
-    }
-}
-
-impl From<Vec<B256>> for BlockTransactions {
-    fn from(hashes: Vec<B256>) -> Self {
-        Self::Hashes(hashes)
-    }
-}
-
-impl<T> From<Vec<T>> for BlockTransactions<T> {
-    fn from(transactions: Vec<T>) -> Self {
-        Self::Full(transactions)
-    }
-}
-
-/// An iterator over the transaction hashes of a block.
-///
-/// See [`BlockTransactions::hashes`].
-#[derive(Clone, Debug)]
-pub struct BlockTransactionHashes<'a, T>(BlockTransactionHashesInner<'a, T>);
-
-#[derive(Clone, Debug)]
-enum BlockTransactionHashesInner<'a, T = Transaction> {
-    Hashes(std::slice::Iter<'a, B256>),
-    Full(std::slice::Iter<'a, T>),
-    Uncle,
-}
-
-impl<'a, T> BlockTransactionHashes<'a, T> {
-    #[inline]
-    fn new(txs: &'a BlockTransactions<T>) -> Self {
-        Self(match txs {
-            BlockTransactions::Hashes(txs) => BlockTransactionHashesInner::Hashes(txs.iter()),
-            BlockTransactions::Full(txs) => BlockTransactionHashesInner::Full(txs.iter()),
-            BlockTransactions::Uncle => BlockTransactionHashesInner::Uncle,
-        })
-    }
-}
-
-impl<'a> Iterator for BlockTransactionHashes<'a, Transaction> {
-    type Item = &'a B256;
-
-    #[inline]
-    fn next(&mut self) -> Option<Self::Item> {
-        match &mut self.0 {
-            BlockTransactionHashesInner::Hashes(txs) => txs.next(),
-            BlockTransactionHashesInner::Full(txs) => txs.next().map(|tx| &tx.hash),
-            BlockTransactionHashesInner::Uncle => None,
-        }
-    }
-
-    #[inline]
-    fn size_hint(&self) -> (usize, Option<usize>) {
-        match &self.0 {
-            BlockTransactionHashesInner::Full(txs) => txs.size_hint(),
-            BlockTransactionHashesInner::Hashes(txs) => txs.size_hint(),
-            BlockTransactionHashesInner::Uncle => (0, Some(0)),
-        }
-    }
-}
-
-impl ExactSizeIterator for BlockTransactionHashes<'_, Transaction> {
-    #[inline]
-    fn len(&self) -> usize {
-        match &self.0 {
-            BlockTransactionHashesInner::Full(txs) => txs.len(),
-            BlockTransactionHashesInner::Hashes(txs) => txs.len(),
-            BlockTransactionHashesInner::Uncle => 0,
-        }
-    }
-}
-
-impl DoubleEndedIterator for BlockTransactionHashes<'_, Transaction> {
-    #[inline]
-    fn next_back(&mut self) -> Option<Self::Item> {
-        match &mut self.0 {
-            BlockTransactionHashesInner::Full(txs) => txs.next_back().map(|tx| &tx.hash),
-            BlockTransactionHashesInner::Hashes(txs) => txs.next_back(),
-            BlockTransactionHashesInner::Uncle => None,
-        }
-    }
-}
-
-impl<'a> std::iter::FusedIterator for BlockTransactionHashes<'a, Transaction> {}
-
-/// An Iterator over the transaction hashes of a block.
-///
-/// See [`BlockTransactions::hashes_mut`].
-#[derive(Debug)]
-pub struct BlockTransactionHashesMut<'a, T = Transaction>(BlockTransactionHashesInnerMut<'a, T>);
```

Comment:
on second thought, we should be able to keep this type?

Reply:
if we'd add `fn tx_hash_mut() -> &mut TxHash` ot `TransactionResponse`, then yes